strawberryfields/apps/subgraph.py

```diff
@@ -271,15 +309,51 @@ def resize(subgraph: list, graph: nx.Graph, min_size: int, max_size: int) -> dic
         shrink_subgraph = graph.subgraph(subgraph).copy()
 
         while shrink_subgraph.order() > min_size:
-            degrees = list(shrink_subgraph.degree())
-            np.random.shuffle(degrees)
+            degrees = np.array(shrink_subgraph.degree)
+            degrees_min = np.argwhere(degrees[:, 1] == degrees[:, 1].min()).flatten()
+
+            if node_select == "uniform":
+                to_remove_index = np.random.choice(degrees_min)
+            elif node_select == "weight":
+                weights = np.array([w[degrees[n][0]] for n in degrees_min])
+                to_remove_index = np.random.choice(np.where(weights == weights.min())[0])
 
-            to_remove = min(degrees, key=lambda x: x[1])
-            shrink_subgraph.remove_node(to_remove[0])
+            to_remove = degrees[to_remove_index][0]
+            shrink_subgraph.remove_node(to_remove)
 
             new_size = shrink_subgraph.order()
 
             if min_size <= new_size <= max_size:
                 resized[new_size] = sorted(shrink_subgraph.nodes())
 
     return resized
+
+
+def _validate_inputs(
```

Comment:
This function probably needs some explanation.

Reply:
Have added more details.

One reason for factoring out this function was because pylint/codefactor complained on the number of `if` statements in `resize`.